### ISSUE TITLE
tag system  suggest

### ISSUE THREAD

**Issue by lbgws2** [zadam/trilium#760]
HI  zadam:

I spent a lot of time looking for a knowledge base software for myself.

When trilium had only 900 stars, I began to pay attention to it,It has many good ideas, clone, run JavaScript in note, save search results, etc.

I'm a collector. I want to store anything I'm interested. It's easy to store, but how to quickly find out what I've collected is a huge problem.For example, I have more than 5000 web urls in my collection,But I don't know how to classify them effectively.

What bothers me is the problem of tree classification. Most of the time, I don't know which classification to put the fragments in. If I forgot which classification to put it in, disaster will come. Unless I use full-text search, I will never find it. And when there are more and more things collected, it's also a frustrating thing to check one by one in the hundreds of search results.

The tag system seems to be able to solve my problem. I'm not a programmer, I try to use VBA to realize it. The interface is ugly, but basically realize my idea. The advantage is that the data is completely stored locally, and the convenience of Excel is incomparable with other applications.

For each collection item, I will input a lot tags on it. There is a certain relationship between these tags: dependency, synonym, attribute, but I will not strictly distinguish them, but just insert them into the note table of SQLite as a record. At the same time, in the tags table, I will divide the tags and use the note ID as joinquery.

When there are enough records, when I want to label a record, I will enter a tag, and the sqlite will find other associated labels, and display the number of times as the sorting basis in list box 2. I can select multiple tags, and then insert them into the tags list of this record. List box 1 is the existing list of tags, which I can exclude by clicking Items.

like this
![image](https://user-images.githubusercontent.com/3014778/70445850-2f2d2500-1ad7-11ea-9e12-d1c2c1ebbaa1.png)

![image](https://user-images.githubusercontent.com/3014778/70446180-b5e20200-1ad7-11ea-84c4-a3cc8b0a5be7.png)

When I want to find a record, the service will filter the result by inputed tag. The more tags I enter, the more accurate the result will be.

Improvement:

1. If no results are found, turn to full-text search

2. Standardize the tags list, such as grouping tags. Enter the name of the group, and the tags under the item will be listed.

Trilium is a great open source software. I think it's better than Evernote. I hope my idea can help it

**Comment by zadam** [comment]
Just a quick check - have you seen attributes (labels) - https://github.com/zadam/trilium/wiki/Attributes

They do support most of what you ask for. There are some missing things (list of existing labels with their usage count) but I imagine that could be provided as a scripted widget. But then I'm not sure if I understood everything completely ...

**Comment by lbgws2** [comment]
Yes, I have noticed the content of attributes. The suggestion is that I want to input a large number of labels quickly. I also notice that all attributes from the template note are inherited to the instance notes.Specifically, I want to be more flexible and intelligent when entering tags, rather than relying on templates alone.
When I input the next tag, it will pop up the suggestion box, list the tags related to last tag, and arrange them according to the count of co-occurrence.

For example, I want to create a Zsh cheatsheet. At the same time, I also have a git cheatsheet.Which path should I put it in, Zsh or Linux command or cheat sheet?

A more in-depth question is how should I create this table, insert a table in note, or create a note for each command

The benefit of create a note for each command is that I can retrieve the tag to get the most accurate result, even in command line with API.
I just want to get what I want quickly, not spend time choosing paths and finding results.

**Comment by zadam** [comment]
BTW thanks for your thoughts, I have been thinking about this since - in general I'm not very happy about the current state of attributes and I might possibly do changes in the direction you described in the 0.39 development ...

**Comment by Vertux** [comment]
@lbgws2 I like the idea of tags very much too. It would be nice to have the option to add tags to e.g. a journal note, it could be a very fast way to assign those daily notes to a couple of subfolders or to show all notes with the same tag by clicking the tag e.g. if you take a journal note of a phone call with a customer you could add a project tag, customer tag and a topic tag (e.g. offer, todo etc.). It would be an additional way to categorize notes very quickly. For me it is a very familiar and comfortable UX - I know it from [Lobsters](https://github.com/jcs/lobsters) and [Wallabag](https://github.com/wallabag/wallabag) - to be honest these tag badges (maybe different colors) look pretty too.

<img width="776" alt="Bildschirm­foto 2023-03-25 um 17 49 41" src="https://user-images.githubusercontent.com/10457918/227731255-0ed6a0ef-016c-4a7b-8539-45ebb8dd1e24.png">

<img width="695" alt="Bildschirm­foto 2023-03-25 um 17 57 37" src="https://user-images.githubusercontent.com/10457918/227731274-b5a8a351-018a-454f-8b06-c0e727250693.png">

<img width="714" alt="Bildschirm­foto 2023-03-25 um 17 51 30" src="https://user-images.githubusercontent.com/10457918/227731286-e38f1810-cc1d-4bf3-9b55-e5305b04616c.png">

**Comment by ZhengfengRao** [comment]
+1

It's a good ideal to list tags/attributes by default, i'm expecting this too

**Comment by lbgws2** [comment]
I found an example about tag filtering

[IMDB keywords](https://www.imdb.com/search/title/?keywords=mafia,murder&explore=keywords)

IMDB has annotated hundreds of keywords for each movie, and by filtering the combination of keywords (i.e. containing multiple keywords at the same time), one can quickly find related movies that interest them

IMDB stores millions of film and television content, and with such a large amount of data, it is very difficult to quickly find the content that interests oneself. Label combination filtering is undoubtedly a very interesting data filtering method
